--- excanvas.js.org	2009-02-10 20:32:35.000000000 +0100
+++ excanvas.js	2009-02-10 20:34:49.000000000 +0100
@@ -39,7 +39,33 @@
 
 
 // only add this code if we do not already have a canvas implementation
-if (!window.CanvasRenderingContext2D) {
+if (!document.createElement('canvas').getContext) {
+
+    (function() {
+	try {
+            if( window.ActiveXObject ) {
+		try {
+                    new ActiveXObject('AgControl.AgControl');
+		    return true;
+		} catch (e) {}
+            } else {
+		if( navigator.plugins["Silverlight Plug-In"] ) {
+		    return true;
+		}
+            }
+	} catch (e) {}
+	
+	if( window.confirm( "Internet Explorer, your browser, " +
+			    "does not support the HTML5 Canvas Tag. " +
+			    "Therefore we need to emulate this using " +
+			    "Silverlight. You browser does not have this " +
+			    "technology installed. Press OK to go the " +
+			    "Silverlight website and install this. Press " +
+			    "Cancel to continue without Silverlight.") )
+	{
+	    window.location.href = "http://silverlight.net";
+	}
+    })();
 
 (function () {
 
